File: Week-1/Solutions/mrprince88/4-fooling-canteen-cashier.py
def solution(n,x,arr):
    even, odd = 0, 0
    
    for i in range(n):
        if arr[i] % 2 == 0:
            even += 1
        else:
            odd += 1
        
    if odd == 0:
        return "No"
    
    else:
        if odd % 2 == 0:
            odd -= 1
        if x % 2 == 0 and even == 0:
            return "No"
        elif even + odd >= x:
            return "Yes"
        else:
            return "No"

File: Week-1/Solutions/shrinivasshah/4-fooling-canteen-cashier.ts
function canChooseFunnySnacks(N: number, X: number, prices: Array<number>): string {
    let odd = prices.filter(p => p % 2 === 1).length;
    let even = N - odd;

    for (let oddCount = 1; oddCount <= Math.min(X, odd); oddCount += 2) {
        let evenCount = X - oddCount;
        if (evenCount <= even) {
            return "YES";
        }
    }
    return "NO";
}

console.log(canChooseFunnySnacks(5, 3, [1, 2, 3, 4, 5]));
console.log(canChooseFunnySnacks(4, 2, [1, 2, 3, 4]));